lib/pathresolver.js

```diff
@@ -86,7 +86,14 @@ PathResolver.prototype = {
     var parsedFrom = url.parse(mainDocUrl);
     var parsedTo = url.parse(absUrl);
     if (parsedFrom.protocol === parsedTo.protocol && parsedFrom.host === parsedTo.host) {
-      var pathname = pathPosix.relative(pathPosix.dirname(parsedFrom.pathname), parsedTo.pathname);
+      var parsedFromDir = pathPosix.dirname(parsedFrom.pathname);
+      // top-level file paths like `index.html` will return the dirname `.`.
+      // To calculate the relative path below, we should resolve against the
+      // root (`/`) instead.
+      if (parsedFromDir === '.') {
```

Comment:
FWIW, the new v3 branch bundler code works around this with a custom URL relative calculation function that incorporates the logic for isAbsoluteUrl checks etc., so I don't want to get too in depth on this fix-- just trying to understand if there is a case where we expect top-level path not to include the leading '/' -- The tests don't seem to have a case for this, so I just wasn't sure how this branch gets exercised...?  Wouldn't this only happen if we were testing relative imports, not absolute imports?  This PR added '/html/absolute-imports.html'.

I feel like I am missing something... HALP!